pkg/dialer/dialer_windows.go

```diff
@@ -32,7 +33,9 @@ func dialer(address string, timeout time.Duration) (net.Conn, error) {
 	return winio.DialPipe(address, &timeout)
 }
 
-// DialAddress returns the dial address
+// DialAddress returns the dial address after trimming off
+// "npipe://" prefix, if any.
 func DialAddress(address string) string {
+	address = strings.TrimPrefix(address, "npipe://")
```

Comment:
The unix counterpart trims the schema in the [`dialer()` function](https://github.com/containerd/containerd/blob/6b7e237fc7f68602b362c89edd0038c7b22a3922/pkg/dialer/dialer_unix.go#L50) before it's used.

It also appends the `unix://` schema in the [`DialAddress()` function](https://github.com/containerd/containerd/blob/6b7e237fc7f68602b362c89edd0038c7b22a3922/pkg/dialer/dialer_unix.go#L33).

If we're to have 1:1 parity here, we should probably do the same, but at this point I think it's enough to just move the `strings.TrimPrrefix()` call in the [`dialer()` function defined above in this file](https://github.com/containerd/containerd/blob/6b7e237fc7f68602b362c89edd0038c7b22a3922/pkg/dialer/dialer_windows.go#L31-L33), before we send that address to `winio.DialPipe()`. I would also pass that address through `filepath.ToSlash()` before trimming the schema.

Also, as a small nit, the `strings.TrimPrefix()` function returns a string. You can skip assigning the output of that function to a variable before returning it. You can simply `return strings.TrimPrefix(address, "npipe://")`.

Reply:
Also, the `DialAddress()` function should probably append the schema if it's not already there, along with the suggested change to trim it in the `dialer()` function. Something like:

```go
address = filepath.ToSlash(address)
if !strings.HasPrefix(address, "npipe://") {
    address = fmt.Sprintf("npipe://%s", address)
}
```

But we'd need to see if this breaks anything.